training/train.py

```diff
@@ -304,113 +304,146 @@ def build_hyperparams(lower_limit, upper_limit, latent_factor,
+    environments = yaml_dict.get('services', [{}])[0].get('environments', [])
+    for index, env in enumerate(environments):
+        if env.get('name', '') == deployment_type:
+            yaml_dict['services'][0]['environments'][index]['comments'] = hyper_params
+            yaml_dict['services'][0]['environments'][index]['parameters']['MODEL_VERSION'] = \
+                model_version
+            break
 
+    return ruamel.yaml.dump(yaml_dict, Dumper=ruamel.yaml.RoundTripDumper)
 
-def get_deployed_model_version(yaml_data, deployment_prefix):
-    """Read deployment yaml and return the deployed model verison."""
-    # 1. Convert yaml to dict
-    yaml_dict = yaml.load(yaml_data)
-
-    # 2. Read model version data for given deploment.
-    model_version = ''
-    environment_name = DEPLOYMENT_PREFIX_ENVIRONMENT_NAME_MAP.get(deployment_prefix, '')
-    if environment_name:
-        environments = yaml_dict.get('services', [{}])[0].get('environments', [])
-        for env in environments:
-            if env.get('name', '') == environment_name:
-                model_version = env.get('parameters', {}).get('MODEL_VERSION', '')
-                break
-    else:
-        _logger.error('ERROR - Invalid deployment prefix: "{}", supported values: {}'.format(
-            deployment_prefix, list(DEPLOYMENT_PREFIX_ENVIRONMENT_NAME_MAP.keys())))
 
-    _logger.info('Model version: {} for deployment prefix: {}'.format(
-        model_version, deployment_prefix))
-    return model_version
+def build_hyper_params_message(hyper_params):
+    """Build hyper params data string used for PR description and in yaml comments."""
+    return '- Hyper parameters :: {}'.format(json.dumps(hyper_params, indent=4, sort_keys=True))
 
 
-def create_git_pr(s3_client, hyper_params):  # pragma: no cover
+def format_body(body):
+    """Format PR body string to replace decorators."""
+    return body.replace('"', '').replace('{', '').replace('}', '').replace(',', '')
+
+
+def read_deployed_data(upstream_repo, s3_client, deployment_type):
+    """Read deployed data like yaml file, hyper params, model version."""
+    upstream_latest_commit_hash = upstream_repo.get_commits()[0].sha
+    _logger.info('Upstream latest commit hash: %s', upstream_latest_commit_hash)
+
+    contents = upstream_repo.get_contents(YAML_FILE_PATH, ref=upstream_latest_commit_hash)
+    yaml_dict = ruamel.yaml.load(contents.decoded_content.decode('utf8'),
+                                 ruamel.yaml.RoundTripLoader)
+
+    deployed_version = get_deployed_model_version(yaml_dict, deployment_type)
+    deployed_file_path = f'{deployed_version}/intermediate-model/hyperparameters.json'
+    deployed_hyperparams = s3_client.read_json_file(deployed_file_path)
+
+    deployed_data = {
+        'version': deployed_version,
+        'hyperparams': deployed_hyperparams
+    }
+    yaml_data = {
+        'commit_hash': upstream_latest_commit_hash,
+        'content_sha': contents.sha,
```

Comment:
Why the upstream commit is associated with yaml data?

Reply:
Created single yaml-data object to represent all associated data. Commit from which it was checked out, its content hash and its content in json dict.

Reply:
By any chance you are storing in yaml? if not don't associate with yaml data